### DISCUSSION TITLE
Improvements and fixes I did in an upcoming PR.

### DISCUSSION THREAD

**Discussion by kasem-sm** [PatilShreyas/NotyKT#330] [Discuss noty-android (Android development)]
**Fixes:**

- Crash when user clicks log out button (simpleapp) (#329 ).
- Fixed view state does not resets after Success or Error state (guided by @PatilShreyas).
- Replaced lifecycleScope.launch with viewLifecycleOwner.lifecycleScope [here](https://github.com/PatilShreyas/NotyKT/blob/master/noty-android/app/simpleapp/src/main/java/dev/shreyaspatil/noty/simpleapp/view/notes/NotesFragment.kt#L278) (probably added by mistake).
- Fix Save Note button hides behind the keyboard (composeapp) by adding windowSoftInputMode attribute in manifest.

---

**Improvements:**

- Added confirmation dialog while logging out and deletion of a note. (simpleapp).
- Fix Save Note button hides behind the keyboard (composeapp).
- wrapped all action Icon composable inside of IconButton at [NotyActions ](https://github.com/PatilShreyas/NotyKT/blob/master/noty-android/app/composeapp/src/main/java/dev/shreyaspatil/noty/composeapp/component/action/NotyActions.kt)(for a better touch region).
- Changed scaffold background color at [AboutScreen ](https://github.com/PatilShreyas/NotyKT/blob/master/noty-android/app/composeapp/src/main/java/dev/shreyaspatil/noty/composeapp/ui/screens/AboutScreen.kt#L101)to match with the background color at [NoteDetailScreen](https://github.com/PatilShreyas/NotyKT/blob/master/noty-android/app/composeapp/src/main/java/dev/shreyaspatil/noty/composeapp/ui/screens/NoteDetailsScreen.kt#L133).
- Changed navigation (back) icon drawable at [AboutScreen ](https://github.com/PatilShreyas/NotyKT/blob/master/noty-android/app/composeapp/src/main/java/dev/shreyaspatil/noty/composeapp/ui/screens/AboutScreen.kt#L95)to match with the icon at [NoteDetailScreen](https://github.com/PatilShreyas/NotyKT/blob/master/noty-android/app/composeapp/src/main/java/dev/shreyaspatil/noty/composeapp/ui/screens/NoteDetailsScreen.kt#L127).
- Changed the text style and padding at [NoteTextFields](https://github.com/PatilShreyas/NotyKT/blob/master/noty-android/app/composeapp/src/main/java/dev/shreyaspatil/noty/composeapp/component/text/NoteTextFields.kt) at composeapp to match the [styles and padding](https://github.com/PatilShreyas/NotyKT/blob/master/noty-android/app/simpleapp/src/main/res/layout/content_note_layout.xml) of simpleapp.
- Refactored [NotyDialogs's ConfirmationDialog ](https://github.com/PatilShreyas/NotyKT/blob/master/noty-android/app/composeapp/src/main/java/dev/shreyaspatil/noty/composeapp/component/dialog/NotyDialogs.kt#L101)to use Default AlertDialog composable instead of Default Dialog composable.

**Comment by PatilShreyas** [comment]
Okay @kasem-sm LGTM

**Reply by PatilShreyas** [reply]
You can raise a PR
